panic: runtime error: invalid memory address or nil pointer dereference when loading org.babashka/fswatcher

## Description

Loading the `org.babashka/fswatcher` pod causes a segmentation fault / nil pointer dereference when running a file, but works finein the REPL.

## Steps to Reproduce

1. Download the correct version of the pod (`0.0.7`)
2. Create a file `main.lg` containing:
   ```clojure
   (pods/load-pod 'org.babashka/fswatcher "0.0.7")
   ```
3. Run the file:
   ```bash
   lg main.lg
   ```

## Expected Behavior

The pod should load successfully without crashing.

## Actual Behavior

The process panics with a nil pointer dereference:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4e4cbd]

goroutine 1 [running]:
github.com/babashka/pod-babashka-fswatcher/babashka.WriteErrorResponse(0x0, {0x54c908?, 0x604c40?})
  /home/circleci/project/babashka/ops.go:81 +0x5d
main.main()
  /home/circleci/project/main.go:12 +0x25


## Notes

- **REPL works fine:** Entering `(pods/load-pod 'org.babashka/fswatcher "0.0.7")` directly into the REPL does **not** trigger the error. This issue appears to be specific to file execution.
- **Pod already downloaded:** Yes, the correct version of the podis already available locally.

## Environment

- **lg version:** lg 1.7.4 (941bbd0)
- **Pod version:** org.babashka/fswatcher 0.0.7
- **OS:** Fedora Linux

--- a/pkg/rt/pods.go
+++ b/pkg/rt/pods.go
@@ -44,7 +44,7 @@ type Pod struct {
 
 	// Response routing
 	pending   map[string]chan podMsg // id -> response channel
-	streaming map[string]bool       // ids that should not auto-close on done
+	streaming map[string]bool        // ids that should not auto-close on done
 	pendingMu sync.Mutex
 	routerUp  bool
 }
@@ -150,7 +150,7 @@ func (p *Pod) startRouter() {
 
 			// Route by id
 			id := bencStr(msg, "id")
-if id == "" {
+			if id == "" {
 				continue
 			}
 
@@ -356,7 +356,13 @@ func (p *Pod) Shutdown() {
 	if p.ops != nil {
 		if _, ok := p.ops["shutdown"]; ok {
 			_ = p.send(map[string]interface{}{"op": "shutdown"})
+			p.stdin.Close()
+			p.cmd.Wait() //nolint:errcheck
+			return
 		}
+	}
+	if p.cmd != nil && p.cmd.Process != nil {
+		_ = p.cmd.Process.Kill()
 	}
 	p.stdin.Close()
 	p.cmd.Wait() //nolint:errcheck
@@ -648,7 +654,7 @@ func installPodsNS() {
 						successFn = hmap.ValueAt(vm.Keyword("success"))
 						errorFn = hmap.ValueAt(vm.Keyword("error"))
 						doneFn = hmap.ValueAt(vm.Keyword("done"))
-						}
+					}
 
 					go func() {
 						for val := range ch {

--- a/pkg/rt/pods_test.go
+++ b/pkg/rt/pods_test.go
@@ -1,0 +1,58 @@
+//go:build !js
+
+/*
+ * Copyright (c) 2026 Marcin Gasperowicz <xnooga@gmail.com>
+ * SPDX-License-Identifier: MIT
+ */
+
+package rt
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+func TestPodShutdownKillsPodWithoutShutdownOp(t *testing.T) {
+	cmd := exec.Command(os.Args[0], "-test.run=TestPodShutdownHelperProcess", "--")
+	cmd.Env = append(os.Environ(), "LET_GO_POD_SHUTDOWN_HELPER=1")
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		t.Fatal(err)
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := cmd.Start(); err != nil {
+		t.Fatal(err)
+	}
+	stderrCh := make(chan []byte, 1)
+	go func() {
+		stderrBytes, _ := io.ReadAll(stderr)
+		stderrCh <- stderrBytes
+	}()
+
+	p := &Pod{
+		cmd:   cmd,
+		stdin: stdin,
+	}
+	p.Shutdown()
+
+	stderrBytes := <-stderrCh
+	if bytes.Contains(stderrBytes, []byte("stdin closed")) {
+		t.Fatalf("expected Shutdown to kill non-shutdown pod before closing stdin; stderr: %s", stderrBytes)
+	}
+}
+
+func TestPodShutdownHelperProcess(t *testing.T) {
+	if os.Getenv("LET_GO_POD_SHUTDOWN_HELPER") != "1" {
+		return
+	}
+	_, _ = io.Copy(io.Discard, os.Stdin)
+	fmt.Fprint(os.Stderr, "stdin closed")
+	os.Exit(0)
+}

